src/common/colorspaces_inline_conversions.h

```diff
@@ -890,7 +890,7 @@ static inline void dt_JzAzBz_2_XYZ(const float *const DT_RESTRICT JzAzBz, float
 
   // X'Y'Z -> XYZ_D65
   XYZ_D65[0] = (XYZ[0] + (b - 1.0f) * XYZ[2]) / b;
-  XYZ_D65[1] = (XYZ[1] + (g - 1.0f) * XYZ[0]) / g;
+  XYZ_D65[1] = (XYZ[1] + (g - 1.0f) * XYZ_D65[0]) / g;
```

Comment:
this does not seem right

Reply:
Cf. Appendix A of reference document.

Inversion of first step:
```
X' = b*X - (b-1)*Z
Y' = g*Y - (g-1)*X
Z' = Z
```
becomes (starting with `Z`, then `X` and finally `Y`; `XYZ` is {`X'`,`Y'`,`Z'`} and `XYZ_D65` is {`X`,`Y`,`Z`}):
```
Z = Z'
X = (X' + (b-1)*Z)/b = (X' + (b-1)*Z')/b
Y = (Y' + (g-1)*X)/g
```

Also by experimentation it gives a better D65 “white” when setting the chroma Cz (or Az and Bz) to zero.

Reply:
Hmm not sure I understand, but I'm pretty busted.

Reply:
@hlecleme is the OpenCL path updated with that ?

Reply:
@aurelienpierre  The function `dt_JzAzBz_2_XYZ` is only used by `gamma.c` which doesn't seem to have an OpenCL version → that function/code is not present in the OpenCL kernel files.

Reply:
Ok thanks. Merging then.